src/NLog/Targets/FileTarget.cs

```diff
@@ -740,11 +739,13 @@ private void RefreshArchiveFilePatternToWatch(string fileName, LogEventInfo logE
             {
                 _fileAppenderCache.CheckCloseAppenders -= AutoCloseAppendersAfterArchive;
 
-                if (KeepFileOpen)
-                    _fileAppenderCache.CheckCloseAppenders += AutoCloseAppendersAfterArchive;
+                bool mustWatchArchiving = IsArchivingEnabled && KeepFileOpen && ConcurrentWrites;
+                if (mustWatchArchiving || (EnableFileDelete && KeepFileOpen && !NetworkWrites && !ReplaceFileContentsOnEachWrite && !EnableFileDeleteSimpleMonitor))
```

Comment:
Could we improve this in terms of readability?

Reply:
Are the conditions duplicated with the properties below?

Reply:
Split into multiple lines.